src/Compilers/Core/CodeAnalysisTest/EditorConfigTests.cs

```diff
@@ -582,7 +589,119 @@ public void WindowsRootConfig()
             {
                 CreateImmutableDictionary(
                     ("cs000", ReportDiagnostic.Suppress))
-            }, options);
+            }, options.TreeOptions);
+        }
+
+        private void VerifyAnalyzerOptions(
+            (string key, string val)[][] expected,
+            AnalyzerConfigOptionsResult options)
+        {
+            var analyzerOptions = options.AnalyzerOptions;
+            Assert.Equal(expected.Length, analyzerOptions.Length);
+
+            for (int i = 0; i < expected.Length; i++)
+            {
+                if (expected[i] is null)
+                {
+                    Assert.Null(analyzerOptions[i]);
+                }
+                else
+                {
+                    AssertEx.SetEqual(
+                        expected[i].Select(KeyValuePair.ToKeyValuePair),
+                        analyzerOptions[i]);
+                }
+            }
+        }
+
+        [Fact]
+        public void SimpleAnalyzerOptions()
+        {
+            var configs = ArrayBuilder<EditorConfig>.GetInstance();
+            configs.Add(EditorConfig.Parse(@"
+[*.cs]
+dotnet_diagnostic.cs000.some_key = some_val", "/.editorconfig"));
+
+            var options = CommonCompiler.GetAnalyzerConfigOptions(
+                new[] { "/test.cs", "/test.vb" },
+                configs,
+                messageProvider: null,
+                diagnostics: null);
+            configs.Free();
+
+            VerifyAnalyzerOptions(
+                new[] {
+                    new[] { ("dotnet_diagnostic.cs000.some_key", "some_val") },
+                    null
```

Comment:
Why does a `null` get generated here?

Reply:
There are two files, so we should expect the outer array to have two elements. The .cs file has a match in the EditorConfig, so it gets the expect values. The second has no matches, so it gets null.